### PR TITLE
Conversion of ordinary least square linear regression (OlsLinearRegression) to estimator

### PR DESCRIPTION
**Pull request by artidoro**
Ongoing work on converting the trainers to estimators. This PR converts the ordinary least squares linear regression (OlsLinearRegression).

### CODE REVIEW

File: src/Microsoft.ML.HalLearners/OlsLinearRegression.cs [hunk 1]
```diff
@@ -65,15 +64,57 @@ public sealed class Arguments : LearnerInputBaseWithWeight
         private static readonly TrainerInfo _info = new TrainerInfo(caching: false);
         public override TrainerInfo Info => _info;
 
-        public OlsLinearRegressionTrainer(IHostEnvironment env, Arguments args)
-            : base(env, LoadNameValue)
+        /// <summary>
+        /// Initializes a new instance of <see cref="OlsLinearRegressionTrainer"/>
+        /// </summary>
+        /// <param name="env">The private instance of <see cref="IHostEnvironment"/>.</param>
```
Comment (Zruty0) [inline src/Microsoft.ML.HalLearners/OlsLinearRegression.cs]: >private [](start = 34, length = 7)

not 'private'. Just say: "The environment to use"

File: src/Microsoft.ML.HalLearners/OlsLinearRegression.cs [hunk 2]
```diff
@@ -65,15 +64,57 @@ public sealed class Arguments : LearnerInputBaseWithWeight
...
+        /// <param name="labelColumn">The name of the label column.</param>
+        /// <param name="featureColumn">The name of the feature column.</param>
+        /// <param name="weightColumn">The name for the column containing the initial weight.</param>
```
Comment (Zruty0) [inline src/Microsoft.ML.HalLearners/OlsLinearRegression.cs]: >column containing the initial weight [](start = 56, length = 36)

example weight column

File: src/Microsoft.ML.HalLearners/OlsLinearRegression.cs [hunk 3]
```diff
@@ -65,15 +64,57 @@ public sealed class Arguments : LearnerInputBaseWithWeight
...
             _l2Weight = args.L2Weight;
             _perParameterSignificance = args.PerParameterSignificance;
         }
 
+        private static Arguments ArgsInit(string featureColumn, string labelColumn,
+            string weightColumn, Action<Arguments> advancedSettings)
+        {
+            var args = new Arguments();
+
+            //apply the advanced args, if the user supplied any
```
Comment (Zruty0) [inline src/Microsoft.ML.HalLearners/OlsLinearRegression.cs]: >//apply the advanced args, if the user supplied any [](start = 12, length = 51)

` // Comments need to be 1 space away from the comment symbol, begin with a capital letter and and with a period.

File: src/Microsoft.ML.HalLearners/OlsLinearRegression.cs [hunk 4]
```diff
@@ -34,7 +33,7 @@
 namespace Microsoft.ML.Runtime.HalLearners
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.HalLearners/OlsLinearRegression.cs]: >Microsoft.ML.Runtime.HalLearners [](start = 10, length = 32)

I'm surprised Pete didn't comment about namespace, especially what it's still "Runtime". But I guess we didn't start to change namespaces for learners yet.

**Pull request merged**